markdown_flow/providers/openai.py

```
@@ -0,0 +1,369 @@
+
+            # Save metadata for retrieval by MarkdownFlow
+            self._last_metadata = metadata
+
+            # Debug output: Response metadata
+            if self.config.debug:
+                self._print_response_metadata(metadata)
+
+            return content
+
+        except Exception as e:
+            raise Exception(f"API request failed: {str(e)}") from e
```

Ruff (0.14.3)

Create your own exception

(TRY002)